### ISSUE TITLE
JSON support

### ISSUE THREAD

**Issue by d1snin** [5HT2/srp-go#2]
You can add support for JSON responses, so your application can be used in other applications, it would be cool

**Comment by d1snin** [comment]
You will probably also have to introduce semantic versioning to maintain backward compatibility

**Comment by 5HT2** [comment]
There is no versioning system in place, and as this program is mostly finished (at least functionality-wise, aside from the TODOs in the README), it doesn't really need one. It's meant to be run from Docker on the latest build. If there is a breaking change in the future, I will do something about it, but I don't think it's reasonable to anticipate it in such a small project.

As for JSON response support... I'm not sure what you mean.

**Comment by d1snin** [comment]
I mean, you can make a full service out of this to get images of various kinds, and use them in other services, I mean, not to return an html page, but the ability to return a json responses.

`localhost:8080/image?json=true`

```json
{
  "status": 200,
  "image": "https://whatever.com/static/image.png"
}
```

**Comment by d1snin** [comment]
it would just be cool in my opinion to customize your application for other needs of this kind and integrate its work into other applications as I said

**Comment by d1snin** [comment]
> There is no versioning system in place, and as this program is mostly finished (at least functionality-wise, aside from the TODOs in the README), it doesn't really need one. It's meant to be run from Docker on the latest build. If there is a breaking change in the future, I will do something about it, but I don't think it's reasonable to anticipate it in such a small project.
> 
> As for JSON response support... I'm not sure what you mean.

this is a different question, but I would not recommend bypassing the use of versioning in docker

**Comment by 5HT2** [comment]
I guess it isn't too hard to just return the JSON data which is normally being generated and stuck into the html.

**Comment by d1snin** [comment]
yes, thanks for your reply.

**Comment by 5HT2** [comment]
Closed by df10cbb41af560c2ad4c21c1026d72314f3a042c
